src/machines/java/com/enderio/machines/common/blockentity/ImpulseHopperBlockEntity.java

```diff
@@ -0,0 +1,104 @@
+package com.enderio.machines.common.blockentity;
+
+import com.enderio.machines.common.MachineTier;
+import com.enderio.machines.common.blockentity.base.PowerConsumingMachineEntity;
+import com.enderio.machines.common.init.MachineCapacitorKeys;
+import com.enderio.machines.common.io.item.MachineInventoryLayout;
+import com.enderio.machines.common.menu.ImpulseHopperMenu;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+
+public class ImpulseHopperBlockEntity extends PowerConsumingMachineEntity{
+    private static final int IMPULSE_HOPPER_POWER_USE_PER_ITEM = 10; //TODO MachineCapacitorKeys.IMPULSE_HOPPER_ENERGY_CONSUME_ITEM
```

Comment:
Don't worry about these capacitor todos. I imagine this PR will merge before #72 so I'll handle these here :)
Not anything needing changed, more of a dev note :)

Reply:
fixed